cmd/zstream/zstream_dump.c

```diff
@@ -215,7 +205,7 @@ sprintf_bytes(char *str, uint8_t *buf, uint_t buf_len)
 }
 
 int
-main(int argc, char *argv[])
+zstream_do_dump(int argc, char *argv[])
```

Comment:
For this case, it worked especially well because zstreamdump was already in its own file, and also the `dump` and `redup` functionalities each have a bunch of code, but don't share much code.  It's probably a less clear win for zfs_main.c / zpool_main.c, but even so it probably would be cleaner to have those broken up into one file per subcommand as well.  Originally we thought that zfs_main.c / zpool_main.c would be pretty thin, with most of the functionality in libzfs.  That's still mostly the case, but a few of the subcommands have grown a bit unwieldy.  This also relates to the proposal for a new, higher-level zfs API: https://openzfs.topicbox.com/groups/developer/Tdde1f0006baa1227-M4c1229e160c31935bc0ff42b